src/lib/ilp/index.js

```diff
@@ -68,17 +61,60 @@ class Ilp {
      * @returns {object} - object containing the fulfilment, ilp packet and condition values
      */
     getQuoteResponseIlp(quoteRequest, quoteResponse) {
-        return this.getResponseIlp({
+        const transactionObject = {
             transactionId: quoteRequest.transactionId,
             quoteId: quoteRequest.quoteId,
             payee: quoteRequest.payee,
             payer: quoteRequest.payer,
             amount: quoteResponse.transferAmount,
             transactionType: quoteRequest.transactionType,
             note: quoteResponse.note
+        };
+        const packetInput = this.makeQuotePacketInput(transactionObject);
+
+        return this.getResponseIlp(transactionObject, packetInput);
+    }
+
+
+    /**
+     * Generates the required fulfilment, ilpPacket and condition for a fxQuote response
+     *
+     * @returns {object} - object containing the fulfilment, ilp packet and condition values
+     */
+    getFxQuoteResponseIlp(fxQuoteRequest, beFxQuoteResponse) {
+        const { conversionRequestId } = fxQuoteRequest;
+        const { conversionTerms } = beFxQuoteResponse;
+        const transactionObject = {
+            conversionRequestId,
+            conversionTerms
+        };
+        const packetInput = this.makeFxQuotePacketInput(transactionObject);
+
+        return this.getResponseIlp(transactionObject, packetInput);
+    }
+
+    makeQuotePacketInput(transactionObject) {
+        return Object.freeze({
+            data: this.makeIlpData(transactionObject), // base64url encoded attached data
+            amount: this._getIlpCurrencyAmount(transactionObject.amount), // unsigned 64bit integer as a string
+            account: this._getIlpAddress(transactionObject.payee) // ilp address
+        });
+    }
+
+    makeFxQuotePacketInput(transactionObject) {
+        return Object.freeze({
+            data: this.makeIlpData(transactionObject),
+            account: this._getFxIlpAddress(transactionObject.conversionTerms), // ilp address
+            amount: this._getIlpCurrencyAmount(transactionObject.conversionTerms.sourceAmount)
+            // todo: discuss what amount we should use for FX: sourceAmount or targetAmount
```

Comment:
Let's discuss this with @PaulGregoryBaker and remove the comment before merging to main.

Reply:
Use `'0'` string as amount for fxQuotes, coz empty string causes an error:
![image](https://github.com/mojaloop/sdk-standard-components/assets/16463615/2fe7c877-0ae2-45b3-a42a-2d9a54a52307)